server/api/plextv.ts

```diff
@@ -87,6 +126,73 @@ class PlexTvAPI {
     });
   }
 
+  public async getDevices(): Promise<PlexDevice[]> {
+    try {
+      const devicesResp = await this.axios.get(
+        '/api/resources?includeHttps=1',
+        {
+          transformResponse: [],
+          responseType: 'text',
+        }
+      );
+      const parsedXml = await xml2js.parseStringPromise(
+        devicesResp.data as DeviceResponse
+      );
+      const devResp = (): PlexDevice[] => {
+        const arr: PlexDevice[] = [];
+        parsedXml?.MediaContainer?.Device?.forEach((pxml: DeviceResponse) => {
+          const connArr: PlexConnection[] = [];
+          pxml?.Connection?.forEach((conn: ConnectionResponse) => {
+            connArr.push({
+              protocol: conn.$.protocol,
+              address: conn.$.address,
+              port: parseInt(conn.$.port, 10),
+              uri: conn.$.uri,
+              local: conn.$.local == '1' ? true : false,
+            });
+          });
```

Comment:
You can simplify this with a map. You can do the same with the actual plex device array as well. This is the preferred pattern in the codebase.

```suggestion
          const connArr: PlexConnection[] = pxml?.Connection?.map((conn: ConnectionResponse) => ({
              protocol: conn.$.protocol,
              address: conn.$.address,
              port: parseInt(conn.$.port, 10),
              uri: conn.$.uri,
              local: conn.$.local == '1' ? true : false,
          });
```